src/server/server.js

```diff
@@ -59,13 +60,14 @@ const startServer = (port, ssoEndpoint, apigeeEndpoint, basePath) => {
                     http.secureProxy(req, res, apigeeEndpoint, basePath, token);
                 },
                 error => {
-                    res.end(`Error: ${error.message}\n`);
+                    res.statusCode = 400;
```

Comment:
Here also I don't know if it is better to consider all errors as user errors, is there a strong reason to do so?

Reply:
Line 60, http.secureProxy() would be happy path flow, where a response is sent to the server and something comes back. That one sets the response code.

I noticed that all the non-happy path cases were not setting the statusCode correctly, so I picked some explicit value, and 400 seemed like the best compromise... except for the authentication case, where I made it 401. But I can do a better job.